### PR TITLE
JENKINS-40472 allow blanks in parameter 'Restrict groups'

### PR DESCRIPTION
**Pull request by gmshake**
Issue link [https://issues.jenkins-ci.org/browse/JENKINS-40472](https://issues.jenkins-ci.org/browse/JENKINS-40472)

### CODE REVIEW

File: src/main/java/de/theit/jenkins/crowd/CrowdConfigurationService.java [hunk 1]
```diff
@@ -110,9 +110,9 @@ public CrowdConfigurationService(String pGroupNames, boolean pNestedGroups) {
 		for (String group : pGroupNames.split(",")) {
 			if (null != group && group.trim().length() > 0) {
```
Comment (oleg-nenashev) [inline src/main/java/de/theit/jenkins/crowd/CrowdConfigurationService.java]: I would rather trim it once before this call

**Comment by gmshake** [comment]
@oleg-nenashev @pingunaut  Done :)

**Comment by KostyaSha** [comment]
removed null check may lead to NPE?

**Comment by gmshake** [comment]
@KostyaSha 
Does not need null check here since tokens return from String.split(String) would never be null in this use case.
```java
for (String group : pGroupNames.split(",")) {
...
}
```
 Tested with Oracle JDK 1.7 and Oracle JDK 1.8.

**Comment by gmshake** [comment]
@pingunaut As the master branch changed a lot since release 1.8, I'd suggest drop minor release 1.9.0 and do a major release 2.0.0

**Pull request merged**